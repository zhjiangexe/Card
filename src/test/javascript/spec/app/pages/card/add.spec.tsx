import React from 'react'
import { shallow, mount } from 'enzyme'
import Add from 'app/pages/card/add'
import sinon from 'sinon'

describe('Card', () => {
  let mountedWrapper
  const devProps = {}
  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(
        <Add addCard={null}/>
      )
    }
    return mountedWrapper
  }

  beforeEach(() => {
    mountedWrapper = undefined
  })

  it('snapshot', () => {
    const component = wrapper()
    expect(component).toMatchSnapshot()
  })
  it('should render correctly in "debug" mode', () => {
    const component = wrapper()
    component.find("[color='primary']").simulate('click')
    expect(component.find('[color="warning"]')).toHaveLength(3)
  })
})

/*
1. import component
2. understanding Enzyme, shallow(淺拷貝), render(HTML), mount(WHOLE DOM TREE)
 */
