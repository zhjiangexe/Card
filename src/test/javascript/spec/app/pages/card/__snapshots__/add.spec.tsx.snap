// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly in "debug" mode 1`] = `
<add
  addCard={null}
>
  <div>
    <h2>
      Add
    </h2>
    <Form
      tag="form"
    >
      <form
        className=""
      >
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Name
              </label>
            </Label>
            <Input
              name="userName"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="userName"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Card number
              </label>
            </Label>
            <Input
              name="cardNo"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="cardNo"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Limit
              </label>
            </Label>
            <Input
              name="limit"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="limit"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Button
              color="primary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Add
              </button>
            </Button>
          </div>
        </FormGroup>
      </form>
    </Form>
  </div>
</add>
`;

exports[`Card snapshot 1`] = `
<add
  addCard={null}
>
  <div>
    <h2>
      Add
    </h2>
    <Form
      tag="form"
    >
      <form
        className=""
      >
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Name
              </label>
            </Label>
            <Input
              name="userName"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="userName"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Card number
              </label>
            </Label>
            <Input
              name="cardNo"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="cardNo"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <label
                className=""
              >
                Limit
              </label>
            </Label>
            <Input
              name="limit"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="form-control"
                name="limit"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </div>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <div
            className="form-group"
          >
            <Button
              color="primary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Add
              </button>
            </Button>
          </div>
        </FormGroup>
      </form>
    </Form>
  </div>
</add>
`;
